src/Microsoft.Repl.Tests/ShellTests.cs

```diff
@@ -159,19 +226,347 @@ public async Task RunAsync_WithUnhandledKeyPress_DoesNothing()
             Assert.Equal(inputBufferText, shell.ShellState.InputManager.GetCurrentBuffer());
         }
 
-        private Shell CreateShell(ConsoleKey consoleKey, int caretPosition, string previousCommand, string nextCommand, out CancellationTokenSource cancellationTokenSource)
+        [Fact]
+        public async Task RunAsync_WithTabKeyPress_UpdatesInputBufferWithFirstEntryFromSuggestionList()
+        {
+            ConsoleKeyInfo consoleKeyInfo = new ConsoleKeyInfo(keyChar: '\0',
+                key: ConsoleKey.Tab,
+                shift: false,
+                alt: false,
+                control: false);
+            Shell shell = CreateShell(consoleKeyInfo,
+                caretPosition: 0,
+                previousCommand: null,
+                nextCommand: null,
+                out CancellationTokenSource cancellationTokenSource);
+
+            string inputBufferTextBeforeKeyPress = "g";
+            string inputBufferTextAfterKeyPress = "get";
+
+            IShellState shellState = shell.ShellState;
+            shellState.InputManager.SetInput(shellState, inputBufferTextBeforeKeyPress);
+
+            DefaultCommandDispatcher<object> defaultCommandDispatcher = shellState.CommandDispatcher as DefaultCommandDispatcher<object>;
+            string commandName = "get";
+            defaultCommandDispatcher.AddCommand(new MockCommand(commandName));
+
+            await shell.RunAsync(cancellationTokenSource.Token).ConfigureAwait(false);
+
+            // Verify the input buffer contents after tab key press event
+            Assert.Equal(inputBufferTextAfterKeyPress, shell.ShellState.InputManager.GetCurrentBuffer());
+        }
+
+        [Fact]
+        public async Task RunAsync_WithShiftTabKeyPress_UpdatesInputBufferWithFirstEntryFromSuggestionList()
```

Comment:
Tab and Shift+Tab do the same thing?

Reply:
Actually it should be first and last. Since I had only one command, it behaved the same. I updated the inputs here to be clear:
https://github.com/aspnet/HttpRepl/commit/c7e2dcc3a354fb3ad79797208be7415e8548b725